library/common-library/templates/_naming.tpl

```diff
@@ -8,6 +8,27 @@ This is an function to be called directly with a string just to truncate strings
 
 
 
+{{- /*
+Given a name and a suffix returns a 'DNS Valid' which always include the suffix, truncating the name if needed.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+It fails if suffix is too long to fint in the limit set.
+Usage:
+{{ include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "<my-name>" "suffix" "my-suffix" ) }}
+*/ -}}
+{{- define "newrelic.common.naming.truncateToDNSWithSuffix" -}}
+{{- $maxLen := (sub 63 (len .suffix)) -}}
+
+{{- if (lt $maxLen 1) -}}
+    {{ fail (printf "The suffix chosen (%s) is hitting the Kubernetes limit of 63 chars for the object name. We cannot create %s-%s" .suffix .name .suffix) }}
```

Comment:
would make sense to warn instead of fail? not sure if there are any implications on this being chopped more than "cosmetic"

Reply:
I think there could be unexpected name collisions (it actually depends on how the helper is used). Normally it would be cosmetic but we cannot be sure...

Reply:
As discussed, failing here is not coherent with truncating long names when there is no suffix, so I changed the helper to truncate not only the name but also the suffix before using it: [94867bc](https://github.com/newrelic/helm-charts/pull/785/commits/94867bcfb9e75d11e708c30a1836001a2d6cb47a)